pkg/simple/client/monitoring/prometheus/prometheus_test.go

```diff
@@ -0,0 +1,90 @@
+package prometheus
+
+import (
+	"kubesphere.io/kubesphere/pkg/simple/client/monitoring"
+	"testing"
+	"time"
+)
+
+func TestGetMetrics(t *testing.T) {
+	tests := []struct {
+		description string
+		queryOption monitoring.QueryOptions
+		expected    *monitoring.MetricsGroup
+	}{
+		{
+			description: "get cluster metrics at a given time",
+			queryOption: monitoring.QueryOptions{
+				Level:         monitoring.MetricsLevelCluster,
+				Start:         time.Unix(1578640245, 0),
+				End:           time.Unix(1578640245, 0),
+				MetricsFilter: "cluster_cpu_usage|cluster_cpu_total",
+			},
+			expected: &monitoring.MetricsGroup{
+				Results: []monitoring.Metrics{
+					{
+						MetricsName: "cluster_cpu_usage",
+						Status:      monitoring.StatusError,
+						Data:        nil,
+						ErrorType:   monitoring.ErrorTypeUnavailable,
+					},
+					{
+						MetricsName: "cluster_cpu_total",
+						Status:      monitoring.StatusError,
+						Data:        nil,
+						ErrorType:   monitoring.ErrorTypeUnavailable,
+					},
+				},
+			},
+		},
+		{
+			description: "get node metrics for a span of time",
+			queryOption: monitoring.QueryOptions{
+				Level:         monitoring.MetricsLevelNode,
+				Start:         time.Unix(1578640245, 0),
+				End:           time.Unix(1578640245, 0),
+				MetricsFilter: "node_cpu_usage|node_cpu_total",
+			},
+			expected: &monitoring.MetricsGroup{
+				Results: []monitoring.Metrics{
+					{
+						MetricsName: "node_cpu_usage",
+						Status:      monitoring.StatusError,
+						Data:        nil,
+						ErrorType:   monitoring.ErrorTypeUnavailable,
+					},
+					{
+						MetricsName: "node_cpu_total",
+						Status:      monitoring.StatusError,
+						Data:        nil,
+						ErrorType:   monitoring.ErrorTypeUnavailable,
+					},
+				},
+			},
+		},
+	}
+
+	opt := &Options{
+		Endpoint:          "http://localhost:9090",
```

Comment:
Isn't this piece of unit test code depending on Prometheus? Then your test code can't run standalone, I think the solution here is to using fake data, or using a mocking Prometheus.